app/controllers/membership_applications_controller.rb

```diff
@@ -47,11 +58,29 @@ def membership_application_params
     params.require(:membership_application).permit(*policy(@membership_application || MembershipApplication).permitted_attributes)
   end
 
+
   def get_membership_application
     @membership_application = MembershipApplication.find(params[:id])
   end
 
+
   def authorize_membership_application
     authorize @membership_application
   end
+
+
+  def new_upload_file(upload_file_param)
+    if upload_file_param['actual_files']
+      upload_file_param['actual_files'].each do |upload_file|
+
+        @uploaded_file = @membership_application.uploaded_files.create(actual_file: upload_file)
+        if @uploaded_file.valid?
+          flash[:notice] = "The file was uploaded: #{@uploaded_file.actual_file_file_name}"
+        else
+          flash[:error] = @uploaded_file.errors.messages
+        end
+      end
+
+    end
+  end
 end
```

Comment:
Agreed.  Should there be a chore for making more complete rspec tests for the MembershipApplicationController?